lisa/analysis/idle.py

```diff
@@ -309,19 +300,17 @@ def plot_clusters_idle_state_residency(self, filepath=None, pct=False):
 
         .. note:: This assumes clusters == frequency domains, which may
           not hold true...
+
+        .. seealso:: :meth:`lisa.analysis.base.AnalysisHelpers.do_plot`
         """
         clusters = self.trace.plat_info['freq-domains']
 
-        fig, axes = self.setup_plot(nrows=len(clusters), sharex=True)
-
-        for idx, cluster in enumerate(clusters):
-            axis = axes[idx]
-
-            self.plot_cluster_idle_state_residency(cluster, pct=pct, axis=axis)
-
-        self.save_plot(fig, filepath)
+        def plotter(axis, local_fig):
+            for idx, cluster in enumerate(clusters):
+                axis = axes[idx]
```

Comment:
```suggestion
                curr_axis = axis[idx]
```
Otherwise that maps to the parameter of the `plot_clusters_*` method

Reply:
Or maybe just ditch the local variable and feed `axis[idx]` straight to the called method...